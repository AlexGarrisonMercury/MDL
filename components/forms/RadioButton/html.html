<div class="radio">
  <input class="radio__input" name ="radio" type="radio" />
</div>
