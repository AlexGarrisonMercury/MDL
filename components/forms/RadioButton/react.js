import React, { memo } from 'react'
import PropTypes from 'prop-types'
import s from './style.module.scss'

const Radio = ({
  checked, onChange, value, name,
}) => (
  <div role="radio" aria-checked={checked} tabIndex={0} className={s.radio}>
    <input
      onClick={event => onChange(event.target.value)}
      onKeyPress={event => onChange(event.target.value)}
      key={value}
      className={s.radio__input}
      type="radio"
      checked={checked}
      value={value}
      readOnly
    />
  </div>
)

Radio.propTypes = {
  checked: PropTypes.bool.isRequired,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  value: PropTypes.string,
  name: PropTypes.string,
}

Radio.defaultProps = {
  disabled: false,
  onChange: null,
  name: 'radio',
}

export default memo(Radio)
