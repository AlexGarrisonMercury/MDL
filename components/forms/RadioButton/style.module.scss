@import "styles/colors.scss";

.radio {
  $size: 14px;
  align-items: center;
  align-content: center;
  position: relative;
  height: $size;
  width: $size;

  &__input {
    position: absolute;
    top: 0;
    left: 0;
    margin: 0;
    cursor: pointer;
    opacity: 0;
  }

  &__icon {
    position: absolute;
    top: 0;
    left: 0;
    content: "";
    background: blue;
    width: 16px;
    height: 16px;
    pointer-events: none;
  }

  &__input:checked ~ &__icon {
    background: red;
  }

  &__svg-wrapper {
    position: relative;
    width: 100%;
    height: 100%;
    pointer-events: none;
  }
}
