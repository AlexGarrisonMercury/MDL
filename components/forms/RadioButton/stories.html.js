import { storiesOf } from '@storybook/html'
import html from './html.html'

storiesOf('Forms/Radio', module).add('default', () => html)
