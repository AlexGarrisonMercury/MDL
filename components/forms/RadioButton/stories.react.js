import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs'
import { withState } from '@dump247/storybook-state'
import Radio from './react'

storiesOf('Forms/Radio', module)
  .addDecorator(withKnobs)
  .add(
    'checked with a click event',
    withState({ checked: true, value: 'true' })(({ store }) => (
      <Radio
        checked={store.state.checked}
        onChange={() => store.set({ checked: !store.state.checked })}
        key={store.state.key}
        name={store.state.name}
        value={store.state.value}
      />
    )),
  )
