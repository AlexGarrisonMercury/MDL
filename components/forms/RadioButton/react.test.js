import React from 'react'
import { mount } from 'enzyme'
import { JSDOM } from 'jsdom'
import RadioButton from './react'

const jsdom = new JSDOM('<!doctype html><html><body><div id="app">Mount here</div></body></html>')
const { window } = jsdom

describe('RadioButton', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = {
      checked: true,
      onChange: jest.fn(),
      value: 'true',
      name: 'radio'
    }

    wrapper = mount(<RadioButton {...props} />, {
      attachTo: window.document.querySelector('#app'),
    })
  })

  describe('with snapshots', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with user actions', () => {
    it('should display the when clicking the radio button', () => {
      wrapper.find('input').simulate('click');
      expect(wrapper.find('input').prop('value')).toEqual('true');
    })
  })
})
