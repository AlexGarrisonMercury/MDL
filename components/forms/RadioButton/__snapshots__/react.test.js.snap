// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton with snapshots should render correctly 1`] = `
<Radio
  checked={true}
  disabled={false}
  name="radio"
  onChange={[MockFunction]}
  value="true"
>
  <div
    aria-checked={true}
    className="radio"
    role="radio"
    tabIndex={0}
  >
    <input
      checked={true}
      className="radio__input"
      key="true"
      onClick={[Function]}
      onKeyPress={[Function]}
      readOnly={true}
      type="radio"
      value="true"
    />
  </div>
</Radio>
`;
